src/main/scala/org/constellation/consensus/RoundManager.scala

```diff
@@ -267,44 +284,42 @@ object RoundManager {
 
   def createRoundData(
     dao: DAO
-  ): Option[(RoundData, Seq[(Transaction, Int)], Seq[(ChannelMessage, Int)])] = {
-
-    val transactions =
-      dao.transactionService
-        .pullForConsensus(dao.minCheckpointFormationThreshold)
-        .unsafeRunSync()
-    if (transactions.nonEmpty) {
-      dao
-        .pullTips(dao.readyFacilitatorsAsync.unsafeRunSync())
-        .map { tips =>
-          val messages = dao.threadSafeMessageMemPool.pull().getOrElse(Seq()) // TODO: Choose more than one tx and light peers
-          val firstTx = transactions.headOption
-          val lightPeers =
-            if (firstTx.isDefined && dao.readyPeers(NodeType.Light).unsafeRunSync().nonEmpty) {
-              Set(
-                dao
-                  .readyPeers(NodeType.Light)
-                  .unsafeRunSync()
-                  .minBy(p => Distance.calculate(firstTx.get.transaction.baseHash, p._1))
-                  ._2
-              )
-            } else Set[PeerData]()
-          val allFacilitators = tips._2.values.map(_.peerMetadata.id).toSet ++ Set(dao.id)
-          (
-            RoundData(
-              generateRoundId,
-              tips._2.values.toSet,
-              lightPeers,
-              FacilitatorId(dao.id),
-              transactions.map(_.transaction),
-              tips._1,
-              messages
-            ),
-            getArbitraryTransactionsWithDistance(allFacilitators, dao).filter(t => t._2 == 1),
-            getArbitraryMessagesWithDistance(allFacilitators, dao).filter(t => t._2 == 1)
-          )
-        }
-    } else None
+  ): Future[Option[(RoundData, Seq[(Transaction, Int)], Seq[(ChannelMessage, Int)])]] = {
+
+    val task = for {
+      transactions <- dao.transactionService.pullForConsensus(dao.minCheckpointFormationThreshold)
+      _ <- if (transactions.isEmpty) IO.raiseError[Unit](NoTransactionsForConsensus) else IO.unit
```

Comment:
if you have `cats.implicits._` imported, you can use `NoTransactionsForConsensus.raiseError[IO, Unit]`, it will be easier for later change from current IO to polymorphic F type